### DISCUSSION TITLE
Automatic permissions for user-owned data

### DISCUSSION THREAD

**Discussion by avimar** [directus/directus#18255] [Feature Requests]
### Summary

We have a button for all permissions to a collection and no permissions.
To create a collection where the user can access only their own data, best to click "all" (to give permission for every field) then add a filter on each one for `user_created = $CURRENT_USER`

It's not that hard to do, but it's repetitive and you have to click the right one.

### Basic Example

I dug into the code and it's not that hard to add this, I have a working proof of concept in less than an hour (first time touching the code base)

1) I duplicated `setFullAccessAll()` 
 as `setFullAccessSelf()` from https://github.com/directus/directus/blob/main/app/src/modules/settings/routes/roles/item/composables/use-update-permissions.ts#L97-L143
and exported it.

Just add in the two permission places:

```
permissions: {
	user_created: {
		_eq: '$CURRENT_USER',
	},
},
```

2) and a new button for it in https://github.com/directus/directus/blob/main/app/src/modules/settings/routes/roles/item/components/permissions-overview-row.vue

3) Add the translations, e.g. label 'self' the the YAML translation files.

It seems to work great!

### Motivation

Easily create collections for self-managed data.

### Detailed Design

Just like the other buttons, it submits permission updates to the API. This will include a limitation for the specified user.

### Must Have:
- Button to give full permission, but limited to access to self

### Drawbacks

- If the user didn't select the optional user_created field, it will either break the SQL or return nothing.
- Currently it duplicates code instead of adding a parameter with the `permissions` limitation.
- It's another button

### Alternatives

It's a manual process that may require looking up the exact syntax.

### Adoption Strategy

If there's no user_created field, then there's no (simple) way to backdate that data.
It theoretically could be automatically added, but old data will not have the proper attribution/permissions.

**Comment by avimar** [comment]
Related question on how to do this at all: https://github.com/directus/directus/discussions/3519

**Comment by rijkvanzanten** [comment]
I'm curious on how common that setup is to know whether or not it warrants one of those "top level" toggle options 🤔 If we'd add "Current User", should we also add one for "Current Role"? I'm wondering if we should maybe move this as a list of "suggested options" above the configuration filter 🤔

**Reply by avimar** [reply]
So maybe a totally different implementation - the ability to duplicate the same permission filters to all create/view/modify/delete/share types.

That's the main issue - it seems for many cases you need to set the exact same thing several times.

**Reply by rijkvanzanten** [reply]
Ah! Good idea. I suppose that would have to be on the drawer level though, as each of those types has different fields. A "Copy from Read" button on the update permissions for example makes a lot of sense to me :)

**Reply by TLMondo** [reply]
Or the ability to Save Custom Rules, and re-use them other places. So in the Custom Rules fly-out, and then the Add Filter, have a section of the custom ones you've saved.

**Reply by SeanDylanGoff** [reply]
i think it could also be a general copy & paste permissions thing, given that it validates the filter before pasting. if a field is missing it ignores the paste and shows an error.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
